src/neural/factory.cc

```diff
@@ -59,7 +62,11 @@ NetworkFactory::Register::Register(const std::string& name, FactoryFunc factory,
 }
 
 void NetworkFactory::PopulateOptions(OptionsParser* options) {
+#ifdef EMBEDDED_WEIGHTS
+  options->Add<StringOption>(NetworkFactory::kWeightsId) = kEmbedded;
```

Comment:
Other options to consider instead of changing the default:
- use the embedded net if autodiscovery fails.
- add the embedded net to the search list (with a timestamp same with the executable) so that newer nets are preferred.

Alternatively (or simultaneously) have a separate (bool) option to use the builtin net.

Reply:
I like the simplicity of you first bullet. If auto-discovery fails then load the embedded. That way the default is kept and also `factory.cc` is left untouched. We lose only one small thing, the ability to specify the embedded weights by config, forcing the user to hide the weights from the auto-discover.

Reply:
I wasn't going for simple, just more intuitive user interface. I expect the `factory.cc` changes should stay to be able to force using the embedded net (or go with an additional option for this).